src/Altinn.App.Core/Helpers/UserHelper.cs

```diff
@@ -81,25 +69,40 @@ public async Task<UserContext> GetUserContext(HttpContext context)
         UserProfile userProfile =
             await _profileClient.GetUserProfile(userContext.UserId)
             ?? throw new Exception("Could not get user profile while getting user context");
+
         userContext.UserParty = userProfile.Party;
 
-        if (context.Request.Cookies[_settings.GetAltinnPartyCookieName] != null)
-        {
-            userContext.PartyId = Convert.ToInt32(
-                context.Request.Cookies[_settings.GetAltinnPartyCookieName],
-                CultureInfo.InvariantCulture
-            );
-        }
+        userContext.PartyId = cookieValue is not null
+            ? Convert.ToInt32(cookieValue, CultureInfo.InvariantCulture)
+            : userContext.PartyId;
 
-        if (userContext.PartyId == userProfile.PartyId)
```

Comment:
This is the check that was altered, re: PR description

Reply:
Why is this changing? I'm not sure I understood what the difference was between these two fields. Have you looked into the Profile API or something?

Reply:
I probably explained that party poorly. The only reason I've changed it is based on the test data we have available. Take for instance the user `1337.json`:

```
{
  "UserId": 1337,
  "UserName": "SophieDDG",
  "PhoneNumber": "90001337",
  "Email": "1337@altinnstudiotestusers.com",
  "PartyId": 501337,
  "Party": {},
  "UserType": 1,
  "ProfileSettingPreference": {
    "Language": "nn",
    "doNotPromptForParty": true
  }
}
```

If `PartyId` for this user matches whatever the `urn:altinn:partyid` (or `AltinnPartyId` cookie) claims, the property `UserContext.Party` will be set to an empty object instead of being looked up with the `IAltinnPartyClient`.

I'm sure this is some form of optimisation, and perhaps an empty `Party` object does not appear in the willd, but in our test environment it definitely produces a faulty logic.

So my proposed solution was to peek inside to the `Party.PartyId` property instead of relying on `PartyId` at root level.

Reply:
Good questions. I don't have the answers here, so perhaps we either need a more thorough investigation, or perhaps I revert the logic and update the test data file instead 🤷

Reply:
As discussed externally, I've rolled back the `Party` assignment to its original logic